src/slices/AccountSlice.ts

```diff
@@ -286,34 +288,49 @@ export const getDonationBalances = createAsyncThunk(
 
       try {
         // NOTE: The BigNumber here is from ethers, and is a different implementation of BigNumber used in the rest of the frontend. For that reason, we convert to string in the interim.
+        const depositIds: BigNumber[] = await givingContract.getDepositorIds(address);
         const allDeposits: [string[], BigNumber[]] = await givingContract.getAllDeposits(address);
-        for (let i = 0; i < allDeposits[0].length; i++) {
-          if (allDeposits[1][i].eq(0)) continue;
-          const depositAmount = ethers.utils.formatUnits(allDeposits[1][i], "gwei");
-          const recipient = allDeposits[0][i];
-          const firstDonationDate: string = await GetDonationDate({
-            address: address,
-            recipient: recipient,
-            networkID: networkID,
-            provider: provider,
-          });
-
-          // NOTE: Bad fix, but since no rebases on testnet this would throw an error otherwise
-          const yieldSent: BigNumber =
-            networkID === 1 ? await givingContract.donatedTo(address, recipient) : BigNumber.from("0");
-          const formattedYieldSent = ethers.utils.formatUnits(yieldSent, "gwei");
-
-          donationInfo.push({
-            date: firstDonationDate,
-            deposit: depositAmount,
-            recipient: recipient,
-            yieldDonated: formattedYieldSent,
-          });
+
+        // ([0x00000...], [0]) is the null return indicating a user has no deposits
+        // This confirms that we did not receive back a null deposit
+        if (allDeposits[0].length != 1 || allDeposits[1][0] != BigNumber.from(0)) {
```

Comment:
Why not flip this and exit the function if the condition isn't satisfied?

Reply:
Is there actually a clean way to do this here? The only way I see is flipping it and then copying up the return from the bottom into that if block, but I'm not sure that's more readable/cleaner since we have (identical looking) returns everywhere

Reply:
probably am just missing something obvious though

Reply:
```
if (allDeposits.length === 0) return;

for (let i = 0; i < allDeposits[0].length; i++) {
   // First entry is of format (etc) if there are no deposits
   if (allDeposits[0][i] == "0x0000 (etc)") break;
   if (allDeposits[1][I].eq(0)) break;

   etc
}
```

Or do the same check out of the for loop:

```
if (allDeposits.length === 0) return;

if (allDeposits[0][0] == "0x0000 (etc)") return;
if (allDeposits[1][0].eq(0)) return;

for (let i = 0; i < allDeposits[0].length; i++) {
```

For the return value, you could define it earlier on so that early returns can use it.